### DISCUSSION TITLE
Separating Members and Stiffening Members

### DISCUSSION THREAD

**Discussion by tbyrk** [idea-statica/ideastatica-public#559] [Rhino/Grasshopper] [answered]
What about completely separating Members and Stiffening Members so that they follow their own logic, data trees and always have their separate inputs and outputs? Right now they are lumped together as Members but their purpose and behaviour is completely different.

When I was making a video about this dome for Linkedin I figured I should not export connection file Members (the 1.25 x h long small members at each connection) to Tekla because in Tekla you actually need the whole member from connection to connection. This would also make the export much faster. But there was no easy fix as unwiring Member export also disabled the export of Stiffening Members because they appear in the same lists. The workaround would have probably been fairly easy to make in this particular case but I can see the same issue arising in many other situations where we are dealing purely with IDEA StatiCa and Tekla is not even in the picture.

Here is an example from my parametric connection generation script. Stiffening Member is listed together with "Connected Members". In IDEA StatiCa Connection application logic they shouldn't be like that because you cannot apply forces on them and they are not part of the global FEA. I'm sure it would be much more practical to have the output "Stiffening Members" just like we have the output "Plates". In this case in "Modify Connection Point" we should also have a separate input "Added Stiffening Member". Yes, the script works just fine as it is right now but I have a gut feeling that the current solution (having Member and Stiffening Members lumped together) will cause issues down the line.

<img width="878" alt="image" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/df70569d-3624-463c-9e83-2c122b94b311">

**Comment by mpospisil** [comment]
Hi,
I agree your proposal to separate input / output makes sense. We  mixed structural members and stiffening members together because it was easiest for implementation (but it causes other problems). 
Can you attach the link to your LinkedIn post? We will pass your proposal to Steel team. There are many other circumstances which need to be taken into account. It would be quite big and expensive change but we should definitely think about it.

Cheers,
Martin

**Comment by tbyrk** [comment]
I understand it is quite a huge change and I'm happy you agree the change makes sense.

[Linkedin](https://www.linkedin.com/posts/tiit-b%C3%BCrkland-614050102_steelconnections-automation-structuralengineering-activity-7204779648340107264-XKlR?utm_source=share&utm_medium=member_desktop)

**Comment by tbyrk** [comment]
Here is one example struggle with Members and Stiffening Members getting mixed together.

https://github.com/idea-statica/ideastatica-public/discussions/491

**Comment by nathanlukeIDEA** [comment]
Hi @tbyrk, 

I can split them in the deconstruct connection point pretty easily however to do it in the program is another story as @mpospisil discusses.

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi @tbyrk, I have updated the Deconstruct Connection Point component to match the Create Connection Point components in terms of inputs and outputs. 
![image](https://github.com/idea-statica/ideastatica-public/assets/93578548/826cc1c2-83b8-4a3d-9615-9ba86a6b0f61)

**Reply by tbyrk** [reply]
Thank for this addition. I will be testing soon.

**Reply by nathanlukeIDEA** [reply]
Make sure you use 0.1.9.

**Reply by tbyrk** [reply]
Looks like updates are coming in daily. Thanks for the reminder.
